src/react-components/auth/AuthContext.js

```diff
@@ -0,0 +1,96 @@
+import React, { createContext, useState, useEffect, useCallback } from "react";
+import PropTypes from "prop-types";
+import configs from "../../utils/configs";
+
+// TODO: We really shouldn't include these dependencies on every page. A dynamic import would work better.
+import jwtDecode from "jwt-decode";
+import AuthChannel from "../../utils/auth-channel";
+import { connectToReticulum } from "../../utils/phoenix-utils";
+
+export const AuthContext = createContext();
+
+export function AuthContextProvider({ children, store }) {
+  const signIn = useCallback(
+    async email => {
+      const authChannel = new AuthChannel(store);
```

Comment:
Seems like we repeat this auth flow a few times throughout the code. I know this part is not new, and I would not suggest refactoring it in this (already large) PR, but it seems we can DRY some of this up.

Reply:
I plan on it in the API PR.